Content.Server/Electrocution/ElectrocutionSystem.cs

```diff
@@ -23,107 +23,137 @@
-
-            UpdatesAfter.Add(typeof(PowerNetSystem));
-        }
+            electrocution.TimeLeft -= timePassed;
+            electrocution.AccumulatedDamage += consumer.ReceivedPower * ElectrifiedDamagePerWatt * timePassed;
 
-        public override void Update(float frameTime)
-        {
-            // Update "in progress" electrocutions
+            if (!MathHelper.CloseTo(electrocution.TimeLeft, 0))
+                continue;
 
-            RemQueue<ElectrocutionComponent> finishedElectrocutionsQueue = new();
-            foreach (var (electrocution, consumer) in EntityManager
-                .EntityQuery<ElectrocutionComponent, PowerConsumerComponent>())
+            if (EntityManager.EntityExists(electrocution.Electrocuting))
             {
-                var ftAdjusted = Math.Min(frameTime, electrocution.TimeLeft);
-
-                electrocution.TimeLeft -= ftAdjusted;
-                electrocution.AccumulatedDamage += consumer.ReceivedPower * ElectrifiedDamagePerWatt * ftAdjusted;
+                // TODO: damage should be scaled by shock damage multiplier
+                // TODO: better paralyze/jitter timing
+                var damage = new DamageSpecifier(_prototypeManager.Index<DamageTypePrototype>(DamageType), (int) electrocution.AccumulatedDamage);
 
-                if (MathHelper.CloseTo(electrocution.TimeLeft, 0))
-                    finishedElectrocutionsQueue.Add(electrocution);
-            }
-
-            foreach (var finished in finishedElectrocutionsQueue)
-            {
-                var uid = finished.Owner;
-                if (EntityManager.EntityExists(finished.Electrocuting))
+                var actual = _damageable.TryChangeDamage(electrocution.Electrocuting, damage, origin: electrocution.Source);
+                if (actual != null)
                 {
-                    // TODO: damage should be scaled by shock damage multiplier
-                    // TODO: better paralyze/jitter timing
-                    var damage = new DamageSpecifier(
-                        _prototypeManager.Index<DamageTypePrototype>(DamageType),
-                        (int) finished.AccumulatedDamage);
-
-                    var actual = _damageableSystem.TryChangeDamage(finished.Electrocuting, damage, origin: finished.Source);
-                    if (actual != null)
-                    {
-                        _adminLogger.Add(LogType.Electrocution,
-                            $"{ToPrettyString(finished.Electrocuting):entity} received {actual.Total:damage} powered electrocution damage from {ToPrettyString(finished.Source):source}");
-                    }
+                    _adminLogger.Add(LogType.Electrocution,
+                        $"{ToPrettyString(electrocution.Electrocuting):entity} received {actual.Total:damage} powered electrocution damage from {ToPrettyString(electrocution.Source):source}");
                 }
+            }
+            EntityManager.DeleteEntity(uid);
```

Comment:
```suggestion
            }
            QueueDel(uid);
```

Iterating entities while deleting them is a sussy proposition.

Reply:
Yeah, before that entities were added to the list first and then were removed, maybe it was actually reasonable.